commandapi-core/src/main/java/dev/jorel/commandapi/network/CommandAPIProtocol.java

```diff
@@ -0,0 +1,72 @@
+package dev.jorel.commandapi.network;
+
+import com.google.common.io.ByteArrayDataInput;
+import dev.jorel.commandapi.network.packets.ClientToServerUpdateRequirementsPacket;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * A utility class that defines the network protocol of the CommandAPI. It handles converting bytes off the network into
+ * {@link CommandAPIPacket}s.
+ */
+public final class CommandAPIProtocol {
+	// Utility class should never be instantiated
+	private CommandAPIProtocol() {}
+
+	/**
+	 * The identifier for the channel used by the CommandAPI to communicate across the network.
+	 */
+	public static final String CHANNEL_NAME = "commandapi:plugin";
+
+	private static final List<Function<ByteArrayDataInput, ? extends CommandAPIPacket>> idToPacket = new ArrayList<>();
+	private static final Map<Class<? extends CommandAPIPacket>, Integer> packetToId = new HashMap<>();
+
+	static {
+		register(ClientToServerUpdateRequirementsPacket.class, ClientToServerUpdateRequirementsPacket::deserialize);
+	}
+
+	/**
+	 * Registers a {@link CommandAPIPacket} so that it can be used on the network. This method assigns ids to each class
+	 * of packet sequentially.
+	 *
+	 * @param clazz        The class of the packet being registered.
+	 * @param deserializer The method that reads this packet from a {@link ByteArrayDataInput}.
+	 */
+	private static <Packet extends CommandAPIPacket> void register(Class<Packet> clazz, Function<ByteArrayDataInput, Packet> deserializer) {
+		if (packetToId.containsKey(clazz)) {
+			throw new IllegalStateException("Packet class \"" + clazz.getSimpleName() + "\" was already registered!");
+		}
+
+		int id = idToPacket.size();
```

Comment:
We don't need this temporary variable, do we? This does the same thing:

```java
packetToId.put(clazz, idToPacket.size());
idToPacket.add(deserializer);
```

Reply:
This is still present in the code 

https://github.com/JorelAli/CommandAPI/blob/a264fecc844cebf2462d706300f24af66743eba8/commandapi-core/src/main/java/dev/jorel/commandapi/network/CommandAPIProtocol.java#L108-L110

I think it might make it clearer that the id is the next one in the sequence, though it's probably not much different.
I'm not sure if either is 'best', I could go either way.